### DISCUSSION TITLE
writeFile of node:fs/promises doesn't work on vercel

### DISCUSSION THREAD

**Discussion by santbob** [vercel/next.js#52044] [Help]
### Summary

I have a `_data` folder in the root of the application and have `.json` files acting as a local db using `node:fs/promises`

I have to write to this file and I use `writeFile` function of `node:fs/promises` it does work in local setup but when deployed in vercel the same is blocked from writing and see the following error.

```
Error writing to the file [Error: EROFS: read-only file system, open '/var/task/_data/app.json'] {
  errno: -30,
  code: 'EROFS',
  syscall: 'open',
  path: '/var/task/_data/app.json'
}
[Error: EROFS: read-only file system, open '/var/task/_data/app.json'] {
  errno: -30,
  code: 'EROFS',
  syscall: 'open',
  path: '/var/task/_data/app.json'
}
[Error: EROFS: read-only file system, open '/var/task/_data/app.json'] {
  errno: -30,
  code: 'EROFS',
  syscall: 'open',
  path: '/var/task/_data/app.json'
}
RequestId: 13ef539a-1115-4841-820d-212a15f45364 Error: Runtime exited with error: exit status 1
Runtime.ExitError

```

here is my write function 

```
lib.update = async (file, data) => {
  let fileHandle;
  try {
    fileHandle = await open(filePath(file), 'r+');
    if (fileHandle) {
      // convert data to string
      const stringData = JSON.stringify(data);
      await fileHandle.truncate()
      await writeFile(fileHandle, stringData);
    } else {
      throw 'Could open the file, it may not exist'
    }
  } catch (e) {
    console.log('Error writing to the file', e)
    throw e
  } finally {
    await fileHandle?.close();
  }
}
```

Is it possible to provide write access to a couple of files or a folder in Vercel

### Additional information

```js
Node - v16.18.1
NextJs - 12.3.0
```

**Comment by icyJoseph** [comment]
One doesn't have write permissions in Vercel deployments, because of the serverless nature of the deployment. 

I'll edit this answer with a few useful links.

- https://github.com/vercel/vercel/discussions/5320#discussioncomment-110775
- https://vercel.com/guides/how-can-i-use-files-in-serverless-functions
